Feature/application entity

ranjeet1515 opened the pull request:
Added database entities.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/yugandhargangu/bettle-track/12)
<!-- Reviewable:end -->

yugandhargangu committed:
Application Directory Structure

1. Bettle track application directory structure.
2. Power point file of directory
3. Gradle script files and sub modules.
4. Spring boot application configuration

yugandhargangu committed:
Update .travis.yml

yugandhargangu committed:
Update .travis.yml

yugandhargangu committed:
Resolved build failure issue.

yugandhargangu committed:
Build fail issue resolved.

yugandhargangu committed:
Build fail issue resolved.

yugandhargangu committed:
Build failed issue patch.

yugandhargangu committed:
Build failure resolved.

yugandhargangu committed:
Review changes

yugandhargangu committed:
Updated with review changes.

yugandhargangu committed:
Added two modules

1. bettle-repo module
   - Contains all repository interfaces
2. bettle-service module
   - COntains all service classes

ranjeet1515 committed:
Added database entities

shashankitmaster commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/common/AppSettings.java~:
```diff
@@ -0,0 +1,35 @@
+package com.bettle.track.entities.common;
```
Please remove this class

shashankitmaster commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Issue.java~:
```diff
@@ -0,0 +1,99 @@
+package com.bettle.track.entities.issue;
```
Please remove this class

shashankitmaster reviewed (changes requested):
Please remove .keep files from repos.

shashankitmaster commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/User.java:
```diff
@@ -0,0 +1,192 @@
+    @Filter(name = "active_dashboard")
+    private Set<Dashboard> listDashboards;
+
+    public Set<Dashboard> getListDashboards() {
```
Remove this getter & setter  if you are using `@Data` in your class.

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Category.java:
```diff
@@ -0,0 +1,29 @@
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class Category extends AbstractParentEntity {
```
Add @FilterDefs for active categories.

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Issue.java:
```diff
@@ -0,0 +1,99 @@
+@Table(name = "m_issues")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue", defaultCondition = "active_flag = 1")
```
Change @FilterDef name from "issue" to "active_issues"

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueAttchment.java:
```diff
@@ -0,0 +1,37 @@
+@Table(name = "")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_attachment", defaultCondition = "active_flag = 1")
```
Change @FilterDef name from "issue_attachment" to "active_issue_attachments".

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueCategory.java:
```diff
@@ -0,0 +1,40 @@
+@Table(name = "m_issues_categories")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issues_categories", defaultCondition = "active_flag = 1")
```
Change @FilterDef name from "issues_categories" to "active_issues_categories".

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueComment.java:
```diff
@@ -0,0 +1,37 @@
+@Table(name = "m_comments")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_comments", defaultCondition = "active_flag = 1")
```
Change @FilterDef from "issue_comments" to "active_issue_comments".

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueDesc.java:
```diff
@@ -0,0 +1,37 @@
+@Table(name = "m_description")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_desc", defaultCondition = "active_flag = 1")
```
Change @FIlterDef name from "issue_desc" to "active_issue_desc"

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueField.java:
```diff
@@ -0,0 +1,45 @@
+@Table(name = "m_issue_custom_fields")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_field", defaultCondition = "active_flag = 1")
```
Change @FilterDef name from "issue_field" to "active_issue_fields".

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueFilter.java:
```diff
@@ -0,0 +1,44 @@
+@Table(name = "m_filters")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "Issue_filter", defaultCondition = "active_flag = 1")
```
Change @FilterDef name from "Issue_filter" to "active_issue_filters".

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueFilterField.java:
```diff
@@ -0,0 +1,44 @@
+@Table(name = "m_filters")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "Issue_filter_field", defaultCondition = "active_flag = 1")
```
Change @FilterDef name from "Issue_filter_field" to "active_issue_filter_fields".

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueFilterSharing.java:
```diff
@@ -0,0 +1,47 @@
+@Table(name = "m_filter_share")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "Issue_filter_sharing", defaultCondition = "active_flag = 1")
```
Change @FilterDef name from "Issue_filter_sharing" to "active_issue_filter_sharing".

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueHistory.java:
```diff
@@ -0,0 +1,39 @@
+@Table(name = "m_issues_history")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_history", defaultCondition = "active_flag = 1")
```
Change @FilterDef name from "issue_history" to "active_issue_history".

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/IssueProcedure.java:
```diff
@@ -0,0 +1,36 @@
+@Table(name = "m_procedure")
+@Where(clause = "active_flag = 1")
+@FilterDefs({
+        @FilterDef(name = "issue_procedures", defaultCondition = "active_flag = 1")
```
Change @FilterDef name from "issue_procedures" to "active_issue_procedures".

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/issue/Tag.java:
```diff
@@ -0,0 +1,30 @@
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class Tag extends AbstractParentEntity {
```
Add @FilterDefs to find active tag records.

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/project/Project.java:
```diff
@@ -0,0 +1,70 @@
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class Project extends AbstractParentEntity {
```
Add @FilterDefs to find active projects.

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/User.java:
```diff
@@ -0,0 +1,192 @@
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class User extends AbstractParentEntity {
```
Add @FilterDefs for active users.

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/UserGroup.java:
```diff
@@ -0,0 +1,43 @@
+@Where(clause = "active_flag = 1")
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class UserGroup extends AbstractParentEntity {
```
Add @FilterDefs for active user groups.

yugandhargangu commented on application/bettle/bettle-entity/src/main/java/com/bettle/track/entities/user/UserInfo.java:
```diff
@@ -0,0 +1,49 @@
+
+@Entity
+@Table(name = "m_user_info")
+@FilterDef(name = "active_user_info", defaultCondition = "active_flag = 1")
```
Add @FilterDefs and keep @FilterDef in @FilterDefs as array.
Add @Where above @FilterDefs annotation.

shashankitmaster commented:
@ranjeet1515 : Please add package-info.java file for your package.

ranjeet1515 committed:
Review changes

1. Removed unnecessary files.
2. Updated @FilterDefs names.
3. Added package-info classes.

ranjeet1515 commented:
Done with all changes.

shashankitmaster reviewed (changes requested):
Please change all `application.properties` to `application.yml`

shashankitmaster reviewed (dismissed):
Add `sourceSet` to `build.gradle`

```java
sourceSets {
    main {
        java {
            srcDirs = ['src/java']
        }
        resources {
            srcDirs = ['src/resources']
        }
    }
}
```

ranjeet1515 committed:
Merge branch 'release' into feature/applicationEntity

yugandhargangu commented:
Reviewed 21 of 54 files at r2, 40 of 40 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 20 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/yugandhargangu/bettle-track/12)*
<!-- Sent from Reviewable.io -->

shashankitmaster commented:
Reviewed 21 of 54 files at r2, 40 of 40 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/yugandhargangu/bettle-track/12)*
<!-- Sent from Reviewable.io -->
